ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
```

Comment:
If we use the LSB of the uint64_t, and shift the vpid to the correct place (so that we logically use the MSB of the jobid), I dont see why this will be an issue if we decide to make the orte_process_name_t a 32 bits value. Why are we trying to increase the opportunities for cache pollution by moving to a design that will require at least 2 extra indirections (the array of arrays solution).

There is a much simpler solution: dissociate the opal_process_name_t from the orte_process_name_t (in a mechanism similar to the lazy naming resolution for shared libraries). Make opal_process_name_t a simple int (32 to start with) that will be the index to an internal array that we alter every time the process known universe changes. This solution might be a little confusing as the index does not represent a global unique naming, but the ORTE name it points to does, but it has the benefit to let us (OPAL and OMPI) be totally independent on how ORTE wants to name it's processes.